opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/NodeLinkRestServiceTestIT.java

```diff
@@ -0,0 +1,462 @@
+        "classpath*:/META-INF/opennms/component-dao.xml",
+        "classpath:/META-INF/opennms/applicationContext-databasePopulator.xml",
+        "classpath:/META-INF/opennms/mockEventIpcManager.xml",
+        "file:src/main/webapp/WEB-INF/applicationContext-svclayer.xml",
+        "file:src/main/webapp/WEB-INF/applicationContext-cxf-common.xml",
+        "classpath:/applicationContext-rest-test.xml"
+})
+@JUnitConfigurationEnvironment
+@JUnitTemporaryDatabase
+public class NodeLinkRestServiceTestIT extends AbstractSpringJerseyRestTestCase {
+    private static final Logger LOG = LoggerFactory.getLogger(NodeLinkRestServiceTestIT.class);
+
+    @Autowired
+    private DatabasePopulator databasePopulator;
+
+    @Autowired
+    private LldpLinkDao lldpLinkDao;
+
+    @Autowired
+    BridgeBridgeLinkDao bridgeBridgeLinkDao;
+
+    @Autowired
+    private CdpLinkDao cdpLinkDao;
+
+    @Autowired
+    private OspfLinkDao ospfLinkDao;
+
+    @Autowired
+    private IsIsLinkDao isisLinkDao;
+
+    @Autowired
+    private LldpElementDao lldpElementDao;
+
+    @Autowired
+    private CdpElementDao cdpElementDao;
+
+    @Autowired
+    private OspfElementDao ospfElementDao;
+
+    @Autowired
+    private IsIsElementDao isisElementDao;
+
+    @Autowired
+    private BridgeElementDao bridgeElementDao;
+
+    public NodeLinkRestServiceTestIT() {
+        super(CXF_REST_V2_CONTEXT_PATH);
+    }
+
+    @Override
+    protected void afterServletStart() throws Exception {
+        MockLogAppender.setupLogging(true, "DEBUG");
+        databasePopulator.populateDatabase();
+    }
+
+    @Test
+    @JUnitTemporaryDatabase
+    @Transactional
+    public void testGetEnlinkd() throws Exception{
+        OnmsNode node1 = createNode1();
+        creatLldpLink(node1);
+        createbridgeBridgeLink(node1);
+        createCdpLink(node1);
+        createOspfLink(node1);
+        createIsIsLink(node1);
+        createLldpElement(node1);
+        createCdpElement(node1);
+        createOspfElement(node1);
+        createIsIsElement(node1);
+
+        String url = "/enlinkd/1";
+        String resultStr = sendRequest(GET, url, 200);
+        LOG.info(resultStr);
+        ObjectMapper mapper = new ObjectMapper();
+        EnlinkdDTO result = mapper.readValue(resultStr, EnlinkdDTO.class);
+        LOG.info(result.toString());
+        Assert.assertEquals(1, result.getLldpLinkNodeDTOs().size());
+        Assert.assertEquals(1, result.getBridgeLinkNodeDTOS().size());
+        Assert.assertEquals(1, result.getCdpLinkNodeDTOS().size());
+        Assert.assertEquals(1, result.getOspfLinkNodeDTOS().size());
+        Assert.assertEquals(1, result.getIsisLinkNodeDTOS().size());
+        Assert.assertNotNull(result.getLldpElementNodeDTO());
+        Assert.assertNotNull(result.getCdpElementNodeDTO());
+        Assert.assertNotNull(result.getOspfElementNodeDTO());
+        Assert.assertNotNull(result.getIsisElementNodeDTO());
+    }
+
+    @Test
```

Comment:
It would be good to add one more test case for failure case.

Reply:
added.